tests/test_date.py

```diff
@@ -297,12 +466,109 @@ def test_should_parse_with_no_break_space_in_dates(self):
 
     @parameterized.expand([
         param(['ur', 'li']),
-        param(['pk',]),
+        param(['pk', ]),
     ])
     def test_should_raise_error_when_unknown_language_given(self, shortnames):
         with self.assertRaisesRegexp(ValueError, '%r' % ', '.join(shortnames)):
             date.DateDataParser(languages=shortnames)
 
+    def when_multiple_dates_are_parsed(self):
+        self.results = []
+        for date_string in self.date_strings:
+            try:
+                result = self.parser.get_date_data(date_string)
+            except ValueError as error:
+                result = error
+            finally:
+                self.results.append(result)
+
+    def given_multiple_dates(self, date_to_parse):
+        self.date_strings = date_to_parse.keys()
+        self.dates = list(date_to_parse.values())
+
+    def given_known_languages(self, *shortnames):
+        self.ordered_languages = OrderedDict()
+
+        self.known_languages = [self.language_map.get(shortname)
+                                for shortname in shortnames]
+
+        for language in self.known_languages:
+            self.ordered_languages[language.shortname] = language
+
+        self.language_loader._data = self.ordered_languages
+
+    def given_empty_languages_parser(self):
+        self._data = {}
+        self.parser = date.DateDataParser(allow_redetect_language=True)
+
+        language_loader = LanguageDataLoader()
+        self.language_map = date.default_language_loader.get_language_map()
+        self.language_loader = language_loader
+        language_loader._load_data = MethodType(self.given_empty_languages_parser, language_loader)
+        self.add_patch(patch('dateparser.date.default_language_loader', new=language_loader))
+
+    def given_date_string(self, date_string):
+        self.date_string = date_string
+
+    def given_date_format(self, date_format):
+        self.date_format = date_format
+
+    def given_expected_date(self, expected_date):
+        self.expected_date = expected_date
+
+    def given_today(self):
+        self.today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+
+    def given_yesterday(self):
+        self.yesterday = self.today - relativedelta(days=1)
+
+    def given_day_before_yesterday(self):
```

Comment:
Are you saying this would replace:

```test_should_return_today```
```test_should_return_yesterday```
```test_should_return_day_before_yesterday```